src/ficamp/sinks/protocols.py

```diff
@@ -0,0 +1,17 @@
+from enum import Enum
+from typing import Protocol
+
+from ficamp.datastructures import Tx
+
+
+class BaseSinkException(Exception):
+    """Base exception for all sink exceptions.
+
+    If your sink function is going to raise an exception, it should inherit from this class.
+    """
+
+    pass
+
+
+class SinkProtocol(Protocol):
+    def write(self, transations: list[Tx]): ...
```

Comment:
not sure how Protocols work. But shouldn't the method `write` take `**kwargs` as well?
I'm thinking about the case that one wants to write to a file (and not only print to stdout). In that case the call would be something like this:

```python
p = SomeSinker()
p.write(transactions, sink=filepath)
```

Reply:
makes sense, thanks!
I was just thinking about a function signature that only depends on inputs (and not state of the class).
But I see your point that if we are anyways instantiating a Sinker we can store attributes in that object and then let the method read them. The only downside I see from your example is that one would need to instantiate a new Sinker for each filepath (or do some logic inside the method to handle more than one filepath).
In any case, I am fine the way the PR is now,  let's merge it and move on – we will see later if we need to adjust anything :)